src/careamics_restoration/engine.py

```diff
@@ -524,3 +530,77 @@ def __del__(self) -> None:
                 if isinstance(handler, FileHandler):
                     self.logger.removeHandler(handler)
                     handler.close()
+
+    def save_as_bioimage(
+        self, output_zip: Union[Path, str], model_specs: Optional[dict] = None
+    ) -> BioimageModel:
+        """Export the current model to BioImage.io model zoo format.
+
+        Parameters
+        ----------
+        output_zip (Union[Path, str]): Where to save the model zip file.
+        model_specs (Optional[dict]): a dictionary that keys are the bioimage-core
+        `build_model` parameters.
+        If None then it will be populated up by the model default specs.
+        """
+        workdir = self.cfg.working_directory
+        # load the best check point
+        checkpoint_path = workdir.joinpath(
+            f"{self.cfg.experiment_name}_best.pth"
+        ).absolute()
+        checkpoint = torch.load(checkpoint_path)
+        # save the model state_dict separately.
+        weight_path = workdir.joinpath("model_weights.pth")
+        torch.save(checkpoint["model_state_dict"], weight_path)
+        # get in/out samples' files
+        test_inputs, test_outputs = self._get_sample_io_files()
+
+        specs = get_default_model_specs(self.cfg.algorithm.loss)
```

Comment:
`algorithm.loss` is a bad proxy for the algorithm name... This will undoubtedly break soon (not all loss will be named after an algorithm, we might also support other losses).

Currently we only support n2v, so I would just simplify everything and then find a better way once we have multiple algorithms.

Reply:
By simplify everything, I mean build the rdf directly for N2V, without trying to support other algorithms. That means no "name" parameter passed to the functions.

Reply:
Yep, agree. I just couldn't find anything else to point to the "N2V".

Reply:
Hard-code it for now, later on we will have criterion based on the Engine to decide the algorithm.

For instance N2V2 has the N2V loss but a different model architecture.